app/controllers/duties_controller.rb

```diff
@@ -150,6 +150,15 @@ def prepare_announcements
   def grab_duty(grab_duty_ids, start_of_week)
     Duty.transaction do
       Duty.find(grab_duty_ids).each do |duty|
+        timeslot = duty.timeslot
+        user_on_duty = Duty.joins(:timeslot)
+                           .where(timeslots: { time_range_id:
+                           timeslot.time_range_id })
+                           .where.not(timeslots: { place_id:
+                           timeslot.place_id })
+                           .exists?(user_id: current_user.id)
+        raise StandardError, 'You need to duty in other places!' if user_on_duty
```

Comment:
Can you move the code to `grabable?` method instead? I feel like it is more suited there. `grab_duty` is for duties that can definitely be grabbed, it is just a driver to executing the database update.

Reply:
Hi Ryo, sorry to reply so late. I have been working on this problem and I have finally solved this. However, there is a little problem that I have discovered for the generate duty function. I have found that in the project's initial method to generate duties, it will generate a schedule where some of the memebers are to duty in different slots at the same time. I think that maybe a new issue could be opened to solve this problem.

Reply:
Dear Ryo, I have tried my best to rewrite and rearrange the method. After spending several hours on it, It can now pass ruboco check on my local machine haha. However, though it seems that the overall coverage has not dropped, I am very sorry that there are still some build problems that I was not able to solve after spending 3 hours on it :sob:

Reply:
Hi Ryo, I have finally solved the problem. I have moved the method into model. I have debugged and now it passed all tests. It will check if the user is dutying at two different places at the same time after running the generate duty or grab duty. It will now pop up warnings if a user is dutying at different places at the same time.